File: framework/src/main/java/com/framework/utils/StringUtils.java
package com.framework.utils;

/**
 * by y on 2017/4/28
 */

public class StringUtils {


    public static String getString(String s, String mark) {
        char[] chars = s.toCharArray();
        StringBuilder buffer = new StringBuilder();
        for (int i = 0; i < chars.length; i++) {
            if (i != chars.length - 1) {
                buffer.append(chars[i]).append(mark);
            } else {
                buffer.append(chars[i]);
            }
        }

        return buffer.toString();
    }


}

File: framework/src/main/java/com/framework/utils/StringUtils.kt
package com.framework.utils

/**
 * by y on 2017/4/28
 */

object StringUtils {


    fun getString(s: String, mark: String): String {
        val chars = s.toCharArray()

        val buffer = StringBuilder()

        for (i in chars.indices) {
            if (i != chars.size - 1) {
                buffer.append(chars[i].toString()).append(mark)
            } else {
                buffer.append(chars[i].toString())
            }
        }

        return buffer.toString()
    }


}
